Develop - 3

grayyyyy456 opened the pull request.

grayyyyy456 committed:
Last commit

grayyyyy456 committed:
добавлен файл requirements.txt

DolgoarshinnykhLG commented on curl.py:
```diff
@@ -0,0 +1,10 @@
+home_page = 'https://stellarburgers.nomoreparties.site/'  # Главная страница
+url_recover_password = 'https://stellarburgers.nomoreparties.site/forgot-password'  # Окно для восстановления пароля
+url_personal_account = 'https://stellarburgers.nomoreparties.site/account/profile'  # Страница профиля после авторизации
+url_order_history = 'https://stellarburgers.nomoreparties.site/account/order-history'  # Страница истории заказов
+url_login = 'https://stellarburgers.nomoreparties.site/login'  # Страница авторизации
+url_feed = 'https://stellarburgers.nomoreparties.site/feed'  # Страница лента заказов
+first_ingredient = 'https://stellarburgers.nomoreparties.site/ingredient/61c0c5a71d1f82001bdaaa6d'  # Страница всплывающего окна первого ингредиента
```
Можно лучше: url-адреса можно конкатенировать с базовым (home_page), так будет проще поддерживать несколько стендов

DolgoarshinnykhLG commented on helper.py:
```diff
@@ -0,0 +1,102 @@
+    counter.wait_button_order()
+    counter.click_button_order()
+    counter.wait_window_exit_order()
+    time.sleep(2)
```
Нужно исправить: во всех тестах лучше обойтись без time.sleep и посмотреть в сторону явных ожиданий selenium`а (expected conditions)

DolgoarshinnykhLG commented on helper.py:
```diff
@@ -0,0 +1,102 @@
+    home_page.wait_button_personal_account()
+    home_page.click_button_personal_account()
+
+@allure.step("Переход в историю заказов")
+def go_to_orders_history(browser):
+    profile_page = ProfilePage(browser)
+    profile_page.wait_orders_button()
+    profile_page.click_orders_button()
+
+@allure.step("Переход в конструктор")
+def go_to_constructor(browser):
+    profile_page = HomePage(browser)
+    profile_page.wait_button_constructor()
+    profile_page.click_button_constructor()
+    return profile_page
+
+@allure.step("Переход в Ленту заказов")
+def go_to_order_feed(browser):
+    profile_page = HomePage(browser)
+    profile_page.wait_button_order_feed()
+    profile_page.click_button_order_feed()
+    return profile_page
+
+@allure.step("Выход из аккаунта")
+def logout(browser):
+    profile_page = ProfilePage(browser)
+    profile_page.wait_exit_button()
+    profile_page.click_exit_button()
+
+
+@allure.step("Процесс создания нового заказа")
+def create_order(browser):
+    counter = OrderPage(browser)
+    counter.wait_first_ingredient()
+    counter.drag_and_drop_ingredient()
+    counter.wait_button_order()
+    counter.click_button_order()
+    counter.wait_window_exit_order()
+    time.sleep(2)
+    counter.click_window_exit_order()
```
Необходимо исправить: в helpers стоит размещать вспомогательные функции (вроде генерации строк), а вызовы для взаимодействия со страницей лучше размещать либо в фикстурах (если это шаг предусловия), либо сразу в тестах

DolgoarshinnykhLG commented on pages/recover_password_page.py:
```diff
@@ -0,0 +1,39 @@
+    @allure.step("Заполняем поле 'email' для восстановления пароля")
+    def filling_email_field(self):
+        self.wait_for_element(RecoverPasswordPageLocators.field_email, timeout=10)
+        email_field = self.find_element(RecoverPasswordPageLocators.field_email)
```
Необходимо исправить: это явно тестовые данные. Все тестовые данные необходимо передавать аргументом функции из тела теста

grayyyyy456 committed:
Конкатенировал урлы

grayyyyy456 committed:
Все тестовые данные передаются аргументом функции из тела теста
Убрал лишние методы из хелпера, переместил в фикстуру и вызывал в тестах
убрал слипы, сколько мог

grayyyyy456 committed:
last commit
